src/JuliusSweetland.OptiKey.Core/Services/WindowManipulationService.cs

```diff
@@ -524,14 +524,17 @@ public void OverridePersistedState(bool inPersistNewState, string inWindowState,
                         ? screenBoundsInDp.Right - newWidth + horizontalOffset
                         : screenBoundsInDp.Width / 2d - newWidth / 2d + horizontalOffset;
                 }
+                if (oldWindowState == WindowStates.Docked)
+                    UnRegisterAppBar();
                 savePersistedState(inPersistNewState);
                 saveFloatingSizeAndPosition(new Rect(newLeft, newTop, newWidth, newHeight));
             }
             else
+            {
+                if (oldWindowState == WindowStates.Docked)
+                    UnRegisterAppBar();
                 savePersistedState(inPersistNewState);
-
-            if (oldWindowState == WindowStates.Docked && newWindowState != WindowStates.Docked)
```

Comment:
@AdamRoden Question: the old logic checks that the new state is not also docked before unregistering the app bar. Is this definitely no longer needed? I.e. is it possible that Optikey changes from docked to docked, but the app bar is unregistered incorrectly?

Thanks!

Reply:
@AdamRoden Ignore that. I've just checked the code in full. All good.